lib/rspec/matchers/built_in/compound.rb

```diff
@@ -101,6 +123,11 @@ def matcher_supports_block_expectations?(matcher)
           false
         end
 
+        def matcher_list_for(matcher)
```

Comment:
Also, if we do update the name to `diffable_matcher_list_from`, we should probably put `matcher.diffable?` check here to match the method name...which would also reduce duplication a bit.